src/NATS.Client.KeyValueStore/Internal/NatsKVWatcher.cs

```diff
@@ -368,6 +367,18 @@ private async ValueTask<INatsJSConsumer> CreatePushConsumer(string origin)
             ReplayPolicy = ConsumerConfigReplayPolicy.Instant,
         };
 
+        // nat2.9.x
+        if (_filters.Length == 1)
+        {
+            config.FilterSubject = _filters[0];
+        }
+
+        // nat2.10.x
+        if (_filters.Length > 1)
+        {
+            config.FilterSubjects = _filters;
+        }
+
```

Comment:
Since it's internal, I thought about throwing an exception in the [Watch()](https://github.com/rickdotnet/nats.net.v2/blob/19da2112b0190454f52e76d59ec011ef75b33204/src/NATS.Client.KeyValueStore/NatsKVStore.cs#L324) method in `NatsKVStore` since that will be closer to the caller. That'd be my vote.

Reply:
I put a check deep down in the watcher class because that's where the enumeration materializes.